src/main/java/com/imsweb/x12/reader/X12Reader.java

```diff
@@ -207,32 +213,40 @@ private void parse(FileType type, Reader reader) throws IOException {
             // cache definitions of loop starting segments
             getLoopConfiguration(_definition.getLoop(), null);
 
-            _dataLoop = new Loop(null);
-            _dataLoop.setSeparators(separators);
             _errors = new ArrayList<>();
 
             String line = scanner.next().trim();
-            while (scanner.hasNext()) {
+            while (scanner.hasNext() || line.length()>0) { 
                 // Determine if we have started a new loop
```

Comment:
This is to allow the loop to process the final line. The original logic is flawed. If you stop on scanner.hasNext being false there is a possibility you missed the last line, because it is being read at the bottom of the loop. So, I read last line of file, i check to see if there is another and I exit. It is easily verified by removing all the extra whitespace at the end of any of the x12 test files and testing for the existence of the IEA segment.

Reply:
I think adding this after the while loop right before the storeData call would be better:
`  if (!line.isEmpty() && !loopLines.contains(line))
                      loopLines.add(line);`

If you just have a new line at the last line of the file you will never break out of this loop. The only way we could break out is too catch the NoSuchElementException you would get from calling scanner.next() but that doesn't seem the proper way to exit the loop.